numaprom/udsink/train.py

```diff
@@ -141,27 +143,68 @@ def train(datums: List[Datum]) -> Responses:
         # TODO change this to just use **composite_keys
         skeys = [payload.composite_keys["namespace"], payload.composite_keys["name"]]
 
-        version = save_model(
-            skeys=skeys, dkeys=["preproc"], model=preproc_clf, artifact_type="sklearn"
-        )
-        _LOGGER.info(
-            "%s - Preproc model saved with skeys: %s with version: %s", payload.uuid, skeys, version
-        )
-
-        version = save_model(skeys=skeys, dkeys=[model_cfg.name], model=anomaly_model)
-        _LOGGER.info(
-            "%s - Model saved with skeys: %s with version: %s", payload.uuid, skeys, version
-        )
-
-        version = save_model(
-            skeys=skeys, dkeys=["thresh"], model=thresh_clf, artifact_type="sklearn"
-        )
-        _LOGGER.info(
-            "%s - Threshold model saved with skeys: %s with version: %s",
-            payload.uuid,
-            skeys,
-            version,
-        )
+        # Save main model
+        model_registry = RedisRegistry(client=redis_client)
+        try:
+            version = model_registry.save(
+                skeys=skeys,
+                dkeys=[model_cfg.name],
+                artifact=anomaly_model,
+                uuid=payload.uuid,
+                train_size=train_df.shape[0],
+            )
+        except RedisRegistryError as err:
+            _LOGGER.exception(
+                "%s - Error while saving Model with skeys: %s, err: %r", payload.uuid, skeys, err
+            )
+        else:
+            _LOGGER.info(
+                "%s - Model saved with skeys: %s with version: %s", payload.uuid, skeys, version
+            )
+        # Save preproc model
+        try:
+            version = model_registry.save(
```

Comment:
Knitpick: Shall we for now have one saving function implemented here that is reusable for : mlmodel, preprocess and postprocess artifact saving.

We can also refactor accordingly when we have a multi0save function in registry.

Reply:
fyi: https://github.com/numaproj/numalogic/issues/196

Reply:
Yes, which is why we need the multi-save functionality. Let's have that and this will be much cleaner.